llama.cpp

```diff
@@ -1453,8 +1446,16 @@ struct llama_model_loader {
         if (use_mmap) {
             cur->data = (uint8_t *) mapping->addr + offs;
         } else {
-            file.seek(offs, SEEK_SET);
-            file.read_raw(cur->data, ggml_nbytes(cur));
+#ifndef _WIN32
+            pread(fileno(file.fp), cur->data, ggml_nbytes(cur), offs);
+#else
+            auto * handle = (HANDLE)_get_osfhandle(_fileno(file.fp));
+            DWORD nread;
+            OVERLAPPED overlapped = {};
+            overlapped.Offset = DWORD(offs);
+            overlapped.OffsetHigh = DWORD(offs << 32);
+            ReadFile(handle, cur->data, ggml_nbytes(cur), &nread, &overlapped);
+#endif
```

Comment:
May be better to move this to `llama_file`. All the calls should be checked for errors, like every other I/O operation. `pread` will fail to read more than 2GB at a time on Linux, so this needs to be considered as well. This is probably already a problem with falcon 180B, with fp32 the output tensor should be around 3.6 GiB if I am not mistaken.